### PR TITLE
⬆️ Upgrade wandb

### PR DESCRIPTION
**Pull request by ashwinvaidya17**
## 📝 Description

- Upgrade wandb https://github.com/openvinotoolkit/anomalib/pull/1928
- Fixes for logger

# Test

image

## ✨ Changes

Select what type of change your PR is:

- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [ ] 🔨 Refactor (non-breaking change which refactors the code base)
- [ ] 🚀 New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📚 Documentation update
- [ ] 🔒 Security update

## ✅ Checklist

Before you submit your pull request, please make sure you have completed the following steps:

- [ ] 📋 I have summarized my changes in the [CHANGELOG](https://github.com/openvinotoolkit/anomalib/blob/main/CHANGELOG.md) and followed the guidelines for my type of change (skip for minor changes, documentation updates, and test enhancements).
- [ ] 📚 I have made the necessary updates to the documentation (if applicable).
- [ ] 🧪 I have written tests that support my changes and prove that my fix is effective or my feature works (if applicable).

For more information about code review checklists, see the [Code Review Checklist](https://github.com/openvinotoolkit/anomalib/blob/main/docs/source/markdown/guides/developer/code_review_checklist.md).

### CODE REVIEW

**Review by samet-akcay — APPROVED** [review]

**Review by samet-akcay — COMMENTED** [review]
Would it be an idea to remove the strict version requirement from `pyproject.toml` file?

**Comment by ashwinvaidya17** [comment]
> Would it be an idea to remove the strict version requirement from `pyproject.toml` file?

Yeah. I think originally the strict requirement was due to HPO features. Since we don't have it anymore we can make it flexible.

**Comment by ashwinvaidya17** [comment]
Need to limit wandb version. See issue: https://github.com/openvinotoolkit/anomalib/issues/2047

File: src/anomalib/deploy/export.py [hunk 1]
```diff
@@ -71,7 +93,8 @@ def __init__(self, model: nn.Module, transform: Transform, disable_antialias: bo
     def forward(self, batch: torch.Tensor) -> torch.Tensor | tuple[torch.Tensor, torch.Tensor]:
         """Transform the input batch and pass it through the model."""
         batch = self.transform(batch)
-        return self.model(batch)
+        with model_eval(self.model):
```
Comment (ashwinvaidya17) [inline src/anomalib/deploy/export.py:96]: @djdameln since you created `InferenceModel` can you have a look at this?

**Pull request merged**